dpctl/tensor/libtensor/source/accumulators/cumulative_prod.cpp

```diff
@@ -138,7 +140,9 @@ struct CumProd1DContigFactory
         if constexpr (TypePairSupportDataForProdAccumulation<srcTy,
                                                              dstTy>::is_defined)
         {
-            using ScanOpT = sycl::multiplies<dstTy>;
+            using ScanOpT = std::conditional_t<std::is_same_v<dstTy, bool>,
+                                               sycl::logical_and<dstTy>,
+                                               sycl::multiplies<dstTy>>;
```

Comment:
It seems the same per every factory.
Should we define an alias for
```cpp
std::conditional_t<std::is_same_v<dstTy, bool>,
                                               sycl::logical_and<dstTy>,
                                               sycl::multiplies<dstTy>>
```
and to reuse in every factory to reduce duplication?

Reply:
Added this change, good suggestion